File: java/src/main/java/com/company/automation/clients/UsersClient.java
package com.company.automation.clients;

import static io.restassured.RestAssured.given;

import com.company.automation.models.RegisterRequest;
import com.company.automation.support.ApiSpec;
import io.restassured.response.Response;

/** User account endpoints. */
public final class UsersClient {

  /** {@code POST /users/register}. Returns the result whatever the status. */
  public ApiResult<Void> register(RegisterRequest request) {
    Response response =
        given().spec(ApiSpec.spec()).body(request).when().post("/users/register").thenReturn();
    return ApiResult.ofStatusOnly(response);
  }
}

File: csharp/src/AutomationFramework.Core/Clients/UsersClient.cs
using System.Net.Http.Json;
using AutomationFramework.Core.Models;
using AutomationFramework.Core.Support;

namespace AutomationFramework.Core.Clients;

/// <summary>User account endpoints.</summary>
public sealed class UsersClient(HttpClient http)
{
    /// <summary><c>POST /users/register</c>. Returns the result whatever the status.</summary>
    public async Task<ApiResult<NoBody>> RegisterAsync(RegisterRequest registerRequest)
    {
        using HttpResponseMessage response = await http
            .PostAsJsonAsync("users/register", registerRequest, ApiHttp.JsonOptions)
            .ConfigureAwait(false);

        return await ApiResult.OfStatusOnlyAsync(response).ConfigureAwait(false);
    }
}
